### PR TITLE
implement Field Commands

### PR DESCRIPTION
**Pull request by SNsir**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 1]
```diff
@@ -15,6 +15,98 @@
...
+  }
+
+  /**
+   * Performs the appropriate action asynchronously.
+   * You can use this method to perform any operation out of the UI event process
+   *
+   * @param    action        the action to perform.
+   * @param    block        This action should block the UI thread ?
+   */
+  override fun performAction(action: Action, block: Boolean) {
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:57]: add annotation : @Deprecated("use method performAsyncAction")

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 2]
```diff
@@ -15,6 +15,98 @@
...
+  override fun performAsyncAction(action: Action) {
+    form.performAsyncAction(action)
+  }
+
+  /**
+   * Performs a void trigger
+   *
+   * @param    type    the number of the trigger
+   */
+  override fun executeVoidTrigger(type: Int) {
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:76]: Rename parameter VKT_Type to match overriden method

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 3]
```diff
@@ -15,6 +15,98 @@
...
+      VForm.CMD_EDITITEM, VForm.CMD_EDITITEM_S ->       //      form.getActiveBlock().getActiveField().getUI().loadItem(VForm.CMD_EDITITEM);
+        form.getActiveBlock().activeField.loadItem(VForm.CMD_EDITITEM)
+      VForm.CMD_NEWITEM ->       //      form.getActiveBlock().getActiveField().getUI().loadItem(VForm.CMD_NEWITEM);
+        form.getActiveBlock().activeField.loadItem(VForm.CMD_EDITITEM)
+    }
+  }
+
+  /**
+   *
+   */
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:89]: Delete

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 4]
```diff
@@ -15,6 +15,98 @@
...
+  // ----------------------------------------------------------------------
+  // HELP HANDLING
+  // ----------------------------------------------------------------------
+  override fun helpOnCommand(help:org.kopi.galite.visual.VHelpGenerator) {
+    if (actor == null) {
+      handler = this
+      actor = form.getDefaultActor(trigger)
+    }
+    if (actor == null) {
+      return
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:107]: actor gets forgetDefaultActor(trigger) which returns VActor non nullable , so the second if doesn't make sense only if VActor was nullable , so recheck

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 5]
```diff
@@ -15,6 +15,98 @@
...
+    }
+    actor!!.helpOnCommand(help)
+  }
+
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:111]: Delete empty line

**Review by ZMarouani — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt (hunk 4 above)
Comment (SNsir) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:107]: seconf if left to keep the same behaviour ( pas de changement de comportement )

**Review by ZMarouani — APPROVED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 6]
```diff
@@ -19,12 +19,13 @@
 package org.kopi.galite.form
 
 abstract class VBlock {
-
   var bufferSize = 0 // max number of buffered records
 
   // dynamic data
   var activeRecord = 0 // current record
 
+  lateinit var activeField : VField
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:27]: it's nullable

File: galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt [hunk 7]
```diff
@@ -15,6 +15,233 @@
...
+import kotlin.reflect.KClass
+
+import org.kopi.galite.base.Query
+import org.kopi.galite.list.VColorColumn
+import org.kopi.galite.list.VListColumn
+import org.kopi.galite.util.base.InconsistencyException
+import org.kopi.galite.visual.VlibProperties
+
+class VColorField(width: Int,
+                  height: Int) : VField(1, 1) {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt:34]: reformat this into one line

File: galite-core/src/main/kotlin/org/kopi/galite/form/VActorField.kt [hunk 8]
```diff
@@ -32,11 +33,7 @@ import org.kopi.galite.visual.VlibProperties
  *
  * This class creates a new actor field instance.
  */
-class VActorField(width: Int, height: Int) : VField(1, 1) {
-
-  // ----------------------------------------------------------------------
-  // IMPLEMENTATION
-  // ----------------------------------------------------------------------
+ class VActorField(width: Int, height: Int) : VField(1, 1) {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VActorField.kt:36]: UNDO this change

File: galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt [hunk 9]
```diff
@@ -19,12 +19,13 @@
 package org.kopi.galite.form
 
 abstract class VBlock {
-
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt:22]: undo this change

File: galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt [hunk 10]
```diff
@@ -15,6 +15,233 @@
...
+
+  /**
+   *
+   */
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt:38]: remove

File: galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt [hunk 11]
```diff
@@ -15,6 +15,233 @@
...
+  override fun getListColumn(): VListColumn = VColorColumn(getHeader(), null, getPriority() >= 0)
+
+  /**
+   * verify that text is valid (during typing)
+   */
+  override fun checkText(s: String): Boolean = true
+
+  /**
+   * verify that value is valid (on exit)
+   * @exception    VException    an exception is raised if text is bad
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt:77]: org.kopi.galite.visual.VException

File: galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt [hunk 12]
```diff
@@ -15,6 +15,233 @@
...
+   */
+
+  /**
+   * Returns a string representation of a date value wrt the field type.
+   */
+  fun formatImage(value: Any): String = "image"
+
+  /**
+   * autofill
+   * @exception   VException    an exception may occur in gotoNextField
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VColorField.kt:226]: org.kopi.galite.visual.VException

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 13]
```diff
@@ -15,6 +15,94 @@
...
+  /**
+   * Performs the appropriate action asynchronously.
+   * You can use this method to perform any operation out of the UI event process
+   *
+   * @param    action        the action to perform.
+   * @param    block        This action should block the UI thread ?
+   */
+  @Deprecated("use method performAsyncAction")
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:55]: `@Deprecated("use method performAsyncAction", ReplaceWith("performAsyncAction(action)"))`

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 14]
```diff
@@ -15,6 +15,94 @@
...
+  override fun executeVoidTrigger(VKT_Type: Int) {
+    when (VKT_Type) {
+      VForm.CMD_AUTOFILL ->
+        form.getActiveBlock().activeField.predefinedFill()
+      VForm.CMD_EDITITEM, VForm.CMD_EDITITEM_S ->
+        form.getActiveBlock().activeField.loadItem(VForm.CMD_EDITITEM)
+      VForm.CMD_NEWITEM ->
+        form.getActiveBlock().activeField.loadItem(VForm.CMD_EDITITEM)
+    }
+  }
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:84]: VKT_Type -> type

File: galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt [hunk 15]
```diff
@@ -15,6 +15,94 @@
...
+        actor = form.getDefaultActor(trigger)
+      }
+      return super.getKey()
+    }
+
+  // ----------------------------------------------------------------------
+  // HELP HANDLING
+  // ----------------------------------------------------------------------
+
+  override fun helpOnCommand(help:org.kopi.galite.visual.VHelpGenerator) {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VFieldCommand.kt:98]: VHelpGenerator

and add import

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VCommand.kt [hunk 16]
```diff
@@ -75,7 +80,7 @@ open class VCommand(private var mode: Int, protected val handler: ActionHandler,
     }
   }
 
-  fun performAction() {
+  open fun performAction() {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VCommand.kt:83]: undo
